cmd/octosql/main.go

```diff
@@ -50,72 +30,19 @@ With OctoSQL you don't need O(n) client tools or a large data analysis system de
 		ctx := context.Background()
 		query := args[0]
 
-		// Configuration
-		cfg, err := config.ReadConfig(configPath)
+		executor, err := octosql.NewOctosqlExecutor()
 		if err != nil {
 			log.Fatal(err)
 		}
-		dataSourceRespository, err := physical.CreateDataSourceRepositoryFromConfig(
-			map[string]physical.Factory{
-				"csv":      csv.NewDataSourceBuilderFactoryFromConfig,
-				"json":     json.NewDataSourceBuilderFactoryFromConfig,
-				"mysql":    mysql.NewDataSourceBuilderFactoryFromConfig,
-				"postgres": postgres.NewDataSourceBuilderFactoryFromConfig,
-				"redis":    redis.NewDataSourceBuilderFactoryFromConfig,
-				"excel":    excel.NewDataSourceBuilderFactoryFromConfig,
-			},
-			cfg,
-		)
-		if err != nil {
-			log.Fatal(err)
-		}
-
-		var out output.Output
-		switch outputFormat {
-		case "table":
-			out = table.NewOutput(os.Stdout, false)
-		case "table_row_separated":
-			out = table.NewOutput(os.Stdout, true)
-		case "json":
-			out = jsonoutput.NewOutput(os.Stdout)
-		case "csv":
-			out = csvoutput.NewOutput(',', os.Stdout)
-		case "tabbed":
-			out = csvoutput.NewOutput('\t', os.Stdout)
-		default:
-			log.Fatal("invalid output type")
-		}
 
-		app := app.NewApp(cfg, dataSourceRespository, out)
-
-		// Parse query
-		stmt, err := sqlparser.Parse(query)
+		err = executor.LoadConfiguration(configPath)
 		if err != nil {
-			log.Fatal("couldn't parse query: ", err)
-		}
-		typed, ok := stmt.(sqlparser.SelectStatement)
-		if !ok {
-			log.Fatalf("invalid statement type, wanted sqlparser.SelectStatement got %v", reflect.TypeOf(stmt))
-		}
-		plan, err := parser.ParseNode(typed)
-		if err != nil {
-			log.Fatal("couldn't parse query: ", err)
-		}
-
-		opts := badger.DefaultOptions("")
-		opts.Dir = ""
-		opts.ValueDir = ""
-		opts.InMemory = true
-		db, err := badger.Open(opts)
-		if err != nil {
-			log.Fatal("couldn't open in-memory badger database: ", err)
+			log.Fatal(err)
 		}
-		stateStorage := storage.NewBadgerStorage(db)
 
-		// Run query
-		err = app.RunPlan(ctx, stateStorage, plan)
+		err = executor.RunQuery(ctx, query)
 		if err != nil {
-			log.Fatal("couldn't run plan: ", err)
+			log.Fatal(err)
```

Comment:
I think providing context which part did the log.Fatal can be useful, so please bring that back. (the "couldn't run plan: ")

Reply:
I will bring it back and do similar things for all the three `log.Fatal` statements.